mlflow/utils/requirements_utils.py

```diff
@@ -167,11 +167,41 @@ def _normalize_package_name(pkg_name):
     return _NORMALIZE_REGEX.sub("-", pkg_name).lower()
 
 
+def _iter_requires(name: str):
```

Comment:
This function mirrors the behavior of `pkg_resources.working_set.by_key.get(norm_pkg_name).requires()`

Reply:
could you link source code of `pkg_resources.working_set.by_key.get(norm_pkg_name).requires()` ?

Reply:
@WeichenXu123 I compared the results between `pkg_resources` and `_iter_requires`.